monitoring/oplog-populator/dashboard.py

```diff
@@ -0,0 +1,335 @@
+    legendValues=['percent', 'value'],
+    pieType='donut',
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.BUCKETS.raw() + ')',
+        legendFormat='{{connector}}',
+    )])
+
+max_pipeline_size = Stat(
+    title='Max Pipeline Size Per Connector',
+    description='Maximum size of the mongo pipeline per connector',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='value',
+    noValue='0',
+    reduceCalc='last',
+    format=UNITS.BYTES,
+    orientation='horizontal',
+    graphMode='none',
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.PIPELINE_SIZE.raw() + ')',
+        legendFormat='{{connector}}',
+    )],
+    thresholds=[
+        Threshold('green', 0, 0.),
+    ])
+
+bucket_additions = TimeSeries(
+    title='Bucket Addition Rate',
+    description='Rate at which buckets are getting added to the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.CONNECTOR_CONFIG(opType="add") + '))',
+        legendFormat='{{connector}}',
+    )])
+
+bucket_removals = TimeSeries(
+    title='Bucket Removal Rate',
+    description='Rate at which buckets are getting removed from the connectors',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.CONNECTOR_CONFIG(opType="delete") + '))',
+        legendFormat='{{connector}}',
+    )])
+
+ack_lag = TimeSeries(
+    title='Acknowledgement Lag By Operation Type',
+    description='Delta between a config change in mongo and it getting processed by the oplogPopulator in seconds',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='sum by(opType) (rate(' + Metrics.ACK_LAG.sum() + '))',
+        legendFormat='{{opType}}',
+    )])
+
+reconfig_lag = TimeSeries(
+    title='Reconfig Lag By Connector',
+    description='Time it takes to reconfigure a connector in seconds',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    unit=UNITS.SECONDS,
+    targets=[Target(
+        expr='sum by(connector) (rate(' + Metrics.RECONFIG_LAG.sum() + '))',
+        legendFormat='{{connector}}',
+    )])
+
+bucket_per_connector = TimeSeries(
+    title='Buckets Per Connector',
+    description='Evolution of buckets per connector',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    targets=[Target(
+        expr='sum by(connector) (' + Metrics.BUCKETS.raw() + ')',
```

Comment:
we may want to use `rate()` in here?
and not sure if/how to aggregate....

alternatively, we already have the rate of add and delete, so tracking the number directly here may not make such sense.... but the deviation/variance may (c.f. ⬆️)

Reply:
We don't really need the rate in this case, as we want to see the number of buckets assigned to each connector. The restart of the pod won't create any issues as the number of assigned buckets will be reset to where it was before.